packages/flutter/lib/src/painting/_network_image_io.dart

```diff
@@ -122,3 +126,37 @@ class NetworkImage extends image_provider.ImageProvider<image_provider.NetworkIm
   @override
   String toString() => '$runtimeType("$url", scale: $scale)';
 }
+
+class _ListenerAwareMultiFrameImageStreamCompleter extends MultiFrameImageStreamCompleter {
+  _ListenerAwareMultiFrameImageStreamCompleter({
+    @required Future<ui.Codec> codec,
+    @required double scale,
+    Stream<ImageChunkEvent> chunkEvents,
+    InformationCollector informationCollector,
+    this.sendChunkEventsController,
+  }) : super(codec: codec, scale: scale, chunkEvents: chunkEvents, informationCollector: informationCollector);
+
+  @override
+  void addListener(ImageStreamListener listener) {
+    if (listener.onChunk != null) {
+      sendChunkEventsController.start();
+      chunkListeners.add(listener);
+    }
+    super.addListener(listener);
+  }
+
+  @override
+  void removeListener(ImageStreamListener listener) {
+    super.removeListener(listener);
+    if (listener.onChunk != null) {
+      chunkListeners.remove(listener);
+
+      if (chunkListeners.isEmpty) {
```

Comment:
Could this use the super-classes `hasListeners` instead of separately keeping track of the listeners again?

Reply:
no, it can't because here we are interested in those listeners that have `onChunk` specified.